packages/ecc-utils-design/custom-elements-manifest.config.js

```diff
@@ -0,0 +1,139 @@
+
+const options = program
+  .option("-o, --outdir <string>")
+  .option("--litelement")
+  .option("--analyze", "", true)
+  .parse()
+  .opts();
+
+const getComponentDocumentation = (tag) =>
+  `https://elixir-cloud-components.vercel.app/design/components/${tag.replace(
+    componentsPrefix,
+    ""
+  )}.html`;
+
+function replace(string, terms) {
+  terms.forEach(({ from, to }) => {
+    string = string?.replace(from, to);
+  });
+
+  return string;
+}
+
+export default {
+  globs: ["src/components/**/*.ts"],
+  exclude: ["**/*.styles.ts", "**/*.test.ts", "index.ts"],
+  plugins: [
+    // Append package data
+    {
+      name: "elixir-cloud-package-data",
+      packageLinkPhase({ customElementsManifest }) {
+        customElementsManifest.package = {
+          name,
+          description,
+          version,
+          author,
+          homepage,
+          license,
+        };
+      },
+    },
+    {
+      name: "shoelace-react-event-names",
+      analyzePhase({ ts, node, moduleDoc }) {
+        switch (node.kind) {
+          case ts.SyntaxKind.ClassDeclaration: {
+            const className = node.name.getText();
+            const classDoc = moduleDoc?.declarations?.find(
+              (declaration) => declaration.name === className
+            );
+
+            classDoc.jsDoc = node.jsDoc
+              ?.map((jsDoc) => jsDoc.getFullText())
+              .join("\n");
+
+            if (classDoc?.events) {
+              classDoc.events.forEach((event) => {
+                event.reactName = `on${pascalCase(event.name)}`;
+                event.eventName = `${pascalCase(event.name)}Event`;
+              });
+            }
+            break;
+          }
+          default:
+            break;
+        }
+      },
+    },
+    {
+      name: "shoelace-translate-module-paths",
+      packageLinkPhase({ customElementsManifest }) {
+        customElementsManifest?.modules?.forEach((mod) => {
+          //
+          // CEM paths look like this:
+          //
+          //  src/components/form/index.ts
+          //
+          // But we want them to look like this:
+          //
+          //  components/form/index.js
+          //
+          const terms = [
+            { from: /^src\//, to: "" }, // Strip the src/ prefix
+            { from: /\.(t|j)sx?$/, to: ".js" }, // Convert .ts to .js
+          ];
+
+          mod.path = replace(mod.path, terms);
+
+          for (const ex of mod.exports ?? []) {
+            ex.declaration.module = replace(ex.declaration.module, terms);
+          }
+
+          for (const dec of mod.declarations ?? []) {
+            if (dec.kind === "class") {
+              for (const member of dec.members ?? []) {
+                if (member.inheritedFrom) {
+                  member.inheritedFrom.module = replace(
+                    member.inheritedFrom.module,
+                    terms
+                  );
+                }
+              }
+            }
+          }
```

Comment:
it modifies how the path component path is saved. The path is used when building the react source code, the way the path is stored initially is invalid and cannot be used for imports and its also not relatively correct since its not generated from the react folder but instead from the cwd.

Reply:
well that's just the name of this CEM plugin. I can change the name but since I just copied it form the shoelace code I wanted to try to give proper attribution and that's what they named the plugin